src/Core/src/Core/ISafeAreaView2.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Microsoft.Maui
+{
+	/// <summary>
+	/// Provides functionality for the Page's SafeAreaInsets that may be changed in the future.
+	/// </summary>
+	/// <remarks>
+	/// This interface is only recognized on the iOS/Mac Catalyst platforms; other platforms will ignore it.
+	/// </remarks>
+	internal interface ISafeAreaView2
+	{
+		/// <summary>
+		/// Internal property for the Page's SafeAreaInsets Thickness that may be changed in the future.
+		/// </summary>
+		internal Thickness SafeAreaInsets { set; }
```

Comment:
Setter-only properties are a bit weird and is not expected. However since this is internal and once we decide how we really want to do this we can always change.